feat: Editorial review for Shared Storage API docs (#30427)
mdn/content: files/en-us/web/api/workletsharedstorage/remainingbudget/index.md

---
title: "WorkletSharedStorage: remainingBudget() method"
short-title: remainingBudget()
slug: Web/API/WorkletSharedStorage/remainingBudget
page-type: web-api-instance-method
status:
  - experimental
browser-compat: api.WorkletSharedStorage.remainingBudget
---

{{APIRef("Shared Storage API")}}{{SeeCompatTable}}

The **`remainingBudget()`** method of the
{{domxref("WorkletSharedStorage")}} interface returns the remaining navigation budget for the current origin.

> The navigation budget is the number of bits of entropy permitted inside a {{htmlelement("fencedframe")}} as a result of {{domxref("WindowSharedStorage.selectURL()")}} calls, per origin, per 24-hour time period. This is not the same as the number of navigations; rather it is based on the number of potential navigations in each call. Each time a `selectURL()` navigation occurs, an amount equal to logarithm base 2 of the number of URL choices is deducted from the corresponding origin's budget.

Reviewer: - There's no "fencedframe" HTML element page. Will this be added as part of fenced frame api?
```suggestion
The navigation budget is the number of entropy bits permitted inside a {{htmlelement("fencedframe")}} due to the {{domxref("WindowSharedStorage.selectURL()")}} calls per origin every 24 hours. This is not the same as the number of navigations; rather, it is based on the number of potential navigations in each call. Each time a `selectURL()` navigation occurs, the budget decreases by the logarithm (base 2) of the number of URL choices from the corresponding origin's budget.
```

Author: Correct; all the fanced frame related pages are being added as part of https://github.com/mdn/content/pull/27781, which I'm intending to create the second draft of straight after I've finished this one.

Author: For the change, I mostly used your suggestion, but tweaked it a bit:

> The navigation budget is the number of entropy bits permitted inside a {{htmlelement("fencedframe")}} due to the {{domxref("WindowSharedStorage.selectURL()")}} calls per origin every 24 hours. This is not the same as the number of navigations; rather, it is based on the number of potential navigations in each call. Each time a `selectURL()` navigation occurs, the corresponding origin's budget decreases by the logarithm (base 2) of the number of URL choices.

Reviewer: This is better, thanks!

Navigation budget is a mechanism designed to limit the rate of leakage of cross-site data to the destination pages navigated to in [fenced frames](/en-US/docs/Web/API/Fenced_Frame_API).

## Syntax

```js-nolint
remainingBudget()
```

### Parameters

None.

### Return value

A {{jsxref("Promise")}} that fulfills with a number representing the remaining navigation budget.

### Exceptions

- {{jsxref("TypeError")}}

>   - : Thrown if the worklet module has not yet been added with {{domxref("Worklet.addModule", "addModule()")}}, or if the calling site does not have the Shared Storage API included in a successful [privacy sandbox enrollment process](/en-US/docs/Web/Privacy/Privacy_sandbox/Enrollment).

Reviewer: ```suggestion
  - : Thrown if the worklet module has not yet been added with {{domxref("Worklet.addModule", "addModule()")}} or if the calling site does not have the Shared Storage API included in a successful [privacy sandbox enrollment process](/en-US/docs/Web/Privacy/Privacy_sandbox/Enrollment).
```
